src/lib/core/hooks/geoConfig.ts

```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react';
+import { StudyEntity } from '../types/study';
+import { findGeolocationNode } from '../utils/geoVariables';
+import { leafletZoomLevelToGeohashLevel } from '../utils/visualization';
+import { GeoConfig } from '../types/geoConfig';
+import { sortBy } from 'lodash';
+
+/**
+ * GeoConfig is a list that corresponds to (only) entities that have geo-variables
+ *
+ * Temporary implementation that looks for a specific combination of variables underneath a single variable tree node.
+ * Assumes that aggregationVariables are in ID order (e.g. geohash_1 has the lowest ID and geohash_6 the highest)
+ *
+ */
+
+export function useGeoConfig(entities: StudyEntity[]): GeoConfig[] {
+  return useMemo(
+    () =>
+      entities
+        .map((entity) => {
+          const geolocationNode = findGeolocationNode(entity);
```

Comment:
A lot of the logic in `findGeolocationNode` is repeated below.

Is there a better way to "filter and map" at the same time?